File: contents/forward_euler_method/code/javascript/euler.js
function solveEuler(timeStep, n){
  const result = [];
  if (n != 0) {
    result[0] = 1;
    for (let i = 1; i < n; ++i) {
      result[i] = result[i-1] - 3 * result[i-1] * timeStep;
    }
  } 
  return result;
}

function checkResult(result, threshold, timeStep) {
  let approx = true;
  for (let i = 0; i < result.length; ++i){
    const solution = Math.exp(-3 * i * timeStep);
    if (Math.abs(result[i] - solution) > threshold) {
      console.log(result[i] + " " + solution);
      approx = false;
    }
  }
  return approx;
}

const timeStep = 0.01;
const n = 100;
const threshold = 0.01;

const result = solveEuler(timeStep, n)
const approx = checkResult(result, threshold, timeStep)

if (approx) {
  console.log("All values within threshold");
} else {
  console.log("Value(s) not in threshold");
}

File: contents/forward_euler_method/code/golang/euler.go
package main

import (
	"fmt"
	"math"
)

func forwardEuler(timeStep float64, n int) []float64 {
	result := make([]float64, n)
	result[0] = 1
	for x := 1; x < n; x++ {
		result[x] = result[x-1] - 3*result[x-1]*timeStep
	}
	return result
}

func check(result []float64, threshold, timeStep float64) bool {
	approx := true
	for x := 0.; int(x) < len(result); x++ {
		solution := math.Exp(-3. * x * timeStep)
		if math.Abs(result[int(x)]-solution) > threshold {
			fmt.Println(result[int(x)], solution)
			approx = false
		}
	}
	return approx
}

func main() {
	timeStep, threshold := .01, .01
	n := 100

	result := forwardEuler(timeStep, n)
	if check(result, threshold, timeStep) {
		fmt.Println("All values within threshold")
	} else {
		fmt.Println("Value(s) not within threshold")
	}
}
